src/components/Calendars/ClientScheduleAppointment.tsx

```diff
@@ -0,0 +1,432 @@
+
+                                    {datepickerValue ? new Date(datepickerValue).toLocaleDateString() : 'Data'}
+                                </div>
+
+                                <div
+                                    className="text-center h-full flex justify-center items-center"
+                                    style={{
+                                        backgroundColor: 'rgb(88, 148, 66)',
+                                    }}
+                                >
+                                    <svg
+                                        className="h-5 w-6 text-gray-300 inline-flex"
+                                        fill="none"
+                                        viewBox="0 0 24 24"
+                                        stroke="white"
+                                    >
+                                        <path
+                                            strokeLinecap="round"
+                                            strokeLinejoin="round"
+                                            strokeWidth="2"
+                                            d="M9 5l7 7-7 7"
+                                        />
+                                    </svg>
+                                </div>
+
+                                <div
+                                    style={{
+                                        backgroundColor: step === 1 ? 'rgb(88, 148, 66)' : 'transparent',
+                                        color: step === 1 ? 'white' : 'black',
+                                    }}
+                                    className="font-skylight text-sm font text-center h-full rounded-r-lg flex-1 flex justify-center items-center"
+                                    onClick={() => {
+                                        if (datepickerValue) setStep(1);
+                                    }}
+                                >
+                                    <svg
+                                        xmlns="http://www.w3.org/2000/svg"
+                                        fill="none"
+                                        viewBox="0 0 24 24"
+                                        strokeWidth="1.5"
+                                        stroke="currentColor"
+                                        className="w-6 h-6 mx-2"
+                                    >
+                                        <path
+                                            strokeLinecap="round"
+                                            strokeLinejoin="round"
+                                            d="M12 6v6h4.5m4.5 0a9 9 0 11-18 0 9 9 0 0118 0z"
+                                        />
+                                    </svg>
+
+                                    {selectedHour ? `${selectedHour}h` : 'Hora'}
+                                </div>
+                            </div>
+
+                            {step === 0 && (
```

Comment:
this could be put on top as a state of the render.

Try to never use ternary statements in the render part of the component if its a big thing like this one. Instead create an object before the return that is something like:

const DATE_SELECTION = 0
const HOUR_SELECTION = 1
const content = {
DATE_SELECTION = <component1>
HOUR_SELECTION = <component2>
}

and then on the render, instead of switching between the two, you do `step === DATE_SELECTION && content[step]` or similar